### PR TITLE
www/nginx: Naxsi whitelist improvements

### PR DESCRIPTION
**Pull request by 8191**
Naxsi whitelist configuration has some drawbacks as elaborated in #2151. Here some improvements:

- Allow specifying multiple IDs for whitelist rules
  - The user can specify _1,2,3_ to whitelist rule IDs 1, 2 and 3 with one single whitelist entry.
  - The user can specify _-1,-2,-3_ to whitelist all rules except rules 1, 2 and 3.
- Improvement in config generation
  - `wl:` rules do not take any `msg:` nor `rx:` or `str:` parts (as these parts are not specified in [Naxsi whitelist documentation](https://github.com/nbs-system/naxsi/wiki/whitelists-bnf)).
  - `id:` rules always generate `rx:` or `str:`. Previous implementation allowed generation of `id:` rules without actual value. I do not see any valid use-case for these.
- Add help texts for Rule Type and Match Type attributes in the UI

### CODE REVIEW

**Comment by 8191** [comment]
@fabianfrz could you please have a look at this potential improvement... Thanks!

File: www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml [hunk 1]
```diff
@@ -530,9 +530,13 @@
           </check001>
         </Constraints>
       </message>
-      <identifier type="IntegerField">
+      <identifier type="TextField">
```
Comment (fabianfrz) [inline www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml:533]: We also have a CSV type which can be used width a select multiple if you want to use that in the frontend. In the Backend you will still get the CSV. This is up to you.

**Comment by 8191** [comment]
@fabianfrz I've rethought the auto-completion topic for the IDs; I think the current behaviour (enter and display numerical IDs only) is fine. Displaying the complete rule name might even be confusing, maybe...

Any other findings/comments from your side?

**Comment by 8191** [comment]
@fabianfrz Any further input from your side? Or can we go for a merge already? 😉

**Comment by 8191** [comment]
@fabianfrz Any chance to merge this for 21.1 if even still possible?

**Review by fabianfrz — APPROVED** [review]

**Comment by fabianfrz** [comment]
looks good from my side.

**Review by fichtner — APPROVED** [review]

**Comment by fichtner** [comment]
Merged, thanks! Sorry for the delay.

**Pull request merged**